src/transformers/tokenization_utils_base.py

```diff
@@ -2179,7 +2179,13 @@ def _get_padding_truncation_strategies(
                     max_length = self.model_max_length
 
         # Test if we have a padding token
-        if padding_strategy != PaddingStrategy.DO_NOT_PAD and (not self.pad_token or self.pad_token_id < 0):
+        if padding_strategy != PaddingStrategy.DO_NOT_PAD \
+            and (
+                not self.pad_token
+                or ((not isinstance(self.pad_token_id, list)) and self.pad_token_id < 0)
+                or (isinstance(self.pad_token_id, list) and all(i < 0 for i in self.pad_token_id))
```

Comment:
How did `self.pad_token_id` become a list? This change should be removed IMO.

Reply:
In CharacterBert the tokens are seen as a sequence of characters, so a padding token becomes a list of ids (zeroes). I put that additional check in `tokenization_utils_base` at a time where I was trying to avoid overriding methods to make the CharacterBertTokenizer work, but in the end, I ended up overriding quite a few of them.

I can remove the check and override the CharacterBert tokenizer instead if you prefer 😊